clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -78,9 +84,12 @@
                 if (handshakeResponse.getHandshakeError() != null) {
                     String errorMessage = "Error in handshake " + handshakeResponse.getHandshakeError();
                     logger.log(LogLevel.Error, errorMessage);
-                    throw new RuntimeException(errorMessage);
+                    RuntimeException exception = new RuntimeException(errorMessage);
+                    handshakeResponseFuture.completeExceptionally(exception);
```

Comment:
What happens if `parseHandshakeResponse` throws because the handshake payload isn't parsable?

Reply:
Similar question for `parseMessage` below. Do we have a test for that?

Reply:
We do have a test for it in parseMessages https://github.com/aspnet/SignalR/blob/cd33755bee1bae855dd0366c3dffcc69d52804f9/clients/java/signalr/src/test/java/com/microsoft/signalr/JsonHubProtocolTest.java#L249-L257